src/main/java/uz/ccrew/flightmanagement/repository/PassengerRepository.java

```diff
@@ -11,11 +12,10 @@
 public interface PassengerRepository extends BasicRepository<Passenger, Long> {
     boolean existsByCustomerId(Long customerId);
 
-    @Query("SELECT DISTINCT p FROM Passenger p " +
-            "JOIN p.customer u " +
-            "JOIN ItineraryReservation ir ON ir.passenger = p " +
-            "JOIN ItineraryLeg il ON il.reservation = ir " +
-            "JOIN Leg l ON l = il.leg " +
-            "WHERE l.flightSchedule.flightNumber = :flightNumber")
-    Page<Passenger> findPassengersWithReservedSeatsOnFlight(String flightNumber, Pageable pageable);
-}
\ No newline at end of file
+    @Query("SELECT p FROM Passenger p WHERE p.passengerId IN " +
+            "(SELECT ir.passenger.passengerId FROM ItineraryReservation ir " +
+            "WHERE ir.reservationId IN " +
+            "(SELECT il.reservation.reservationId FROM ItineraryLeg il " +
+            "WHERE il.leg.flightSchedule.flightNumber = :flightNumber))")
```

Comment:
in ham keremas hich narsa ishlatmasdan prosta ichma ich kirib ketsa boladi hibernate enity imkoniyati bor. keyin bu query ReservatinRepository da bolishi kere